cognite_toolkit/cdf.py

```diff
@@ -278,43 +255,20 @@ def deploy(
         )
         exit(1)
     print(ToolGlobals.as_string())
-    if CDFDataTypes.raw in include and Path(f"{build_dir}/raw").is_dir():
-        # load_raw() will assume that the RAW database name is set like this in the filename:
-        # <index>.<raw_db>.<tablename>.csv
-        load_raw(
-            ToolGlobals,
-            raw_db="default",
-            drop=drop,
-            file=None,
-            dry_run=dry_run,
-            directory=f"{build_dir}/raw",
-        )
-        if ToolGlobals.failed:
-            print("[bold red]ERROR: [/] Failure to load RAW as expected.")
-            exit(1)
-    if CDFDataTypes.timeseries in include and Path(f"{build_dir}/timeseries").is_dir():
-        load_timeseries_metadata(
-            ToolGlobals,
-            drop=drop,
-            file=None,
-            dry_run=dry_run,
-            directory=f"{build_dir}/timeseries",
-        )
-        if ToolGlobals.failed:
-            print("[bold red]ERROR: [/] Failure to load timeseries as expected.")
-            exit(1)
-    if CDFDataTypes.transformations in include and Path(f"{build_dir}/transformations").is_dir():
-        load_transformations(
-            ToolGlobals,
-            file=None,
-            drop=drop,
-            dry_run=dry_run,
-            directory=f"{build_dir}/transformations",
-        )
-        if ToolGlobals.failed:
-            print("[bold red]ERROR: [/] Failure to load transformations as expected.")
-            exit(1)
-    if CDFDataTypes.data_models in include and (models_dir := Path(f"{build_dir}/data_models")).is_dir():
+
+    for folder_name, LoaderCls in LOADER_BY_FOLDER_NAME.items():
+        if folder_name in include and (directory := (Path(build_dir) / folder_name)).is_dir():
+            load_resources(
```

Comment:
Running with `./cognite_toolkit/cdf.py deploy --drop --drop-data --env=local ./build ` I get a failure on timeseries as data_set_id in ToolGlobals is None. Line 164 in load.py

Reply:
`  File "/Users/wedel/src/cdf-project-templates/cognite_toolkit/cdf_tk/load.py", line 164, in get_required_capability
    TimeSeriesAcl.Scope.DataSet([ToolGlobals.data_set_id]),
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<string>", line 4, in __init__
  File "/Users/wedel/src/cdf-project-templates/.venv/lib/python3.11/site-packages/cognite/client/data_classes/capabilities.py", line 330, in __post_init__
    object.__setattr__(self, "ids", [int(i) for i in self.ids])
                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/wedel/src/cdf-project-templates/.venv/lib/python3.11/site-packages/cognite/client/data_classes/capabilities.py", line 330, in <listcomp>
    object.__setattr__(self, "ids", [int(i) for i in self.ids])
                                     ^^^^^^
TypeError: int() argument must be a string, a bytes-like object or a real number, not 'NoneType'`

Reply:
See it. An argument in favor of `mypy` which would have caught this one.